### ISSUE TITLE
Issue compiling on Gentoo (CEGUI, FindCEGUI)

### ISSUE THREAD

**Issue by alkavan** [Revolutionary-Games/Thrive#299]
Trying to build the game on Gentoo system.
First, I found there's a dependency package called ```dev-games/ois-1.3``` not mentioned in README.md.

I added to ```cmake_modules/FindCEGUI.cmake``` two paths so the script will find CEGUI on my system as it failed to find it by default.

```
    SET( CEGUI_INCLUDE_SEARCH_DIR
       ...
       /usr/include/cegui-0
    )
```

And also:
```
    SET( CEGUI_LIBRARY_SEARCH_DIR
            ...
            /usr/lib64
    )
```

I also had to comment out conditions related to file names. Case with ```CEGUIBase-0``` and ```CEGUIBase-0_d``` for example, it was looking for ```CEGUIBase-0_d``` and I don't have such file (I guess it's a debug version).

Eventually after all this, configuration stage pass, but unfortunately I still get a compile error while running ```make``` and I'm not sure why:

```
./src/engine/engine.cpp.o: 21:25: fatal error: CEGUI/CEGUI.h: No such file or directory #include <CEGUI/CEGUI.h>
```

Am I doing something wrong? The actual file is here:
```
/usr/include/cegui-0/CEGUI/CEGUI.h
```

**Comment by Moopli** [comment]
My current build setup uses an ugly kludge, `sudo mv /usr/local/include/cegui-0/CEGUI /usr/local/include/CEGUI`. Try adding `/usr/local/include/cegui-0` to `FindCEGUI.cmake:CEGUI_INCLUDE_SEARCH_DIR`.

**Comment by alkavan** [comment]
I just successfully built it.
Only one extra change was need, I had to add the following GCC environment variable:
```
CPLUS_INCLUDE_PATH="/usr/include/cegui-0"
```

I'm not entirely sue if this is an issue with Gentoo's package location or the supplied FindCEGUI.cmake script. Perhaps the script needs some adjustments?

**Comment by hhyyrylainen** [comment]
I had a similar issue which I solved by using the official CEGUI find script from: https://bitbucket.org/cegui/cegui/src/b819d0566f1220e710d55e0600fc4b19b38e4f07/cmake/FindCEGUI.cmake?at=default
Only other change I had to do was adding 
```
    ${CEGUI_OgreRenderer_LIBRARIES}
```
to the list of libraries which ThriveLib links against in the top level `CMakeLists.txt`
```
    target_link_libraries(ThriveLib ...)
```

**Comment by jjonj** [comment]
This is my fault for adding some windows specific directory renaming when I added CEGUI sorry!

**Comment by alkavan** [comment]
great thanks, this seems like my initial issue not finding: ```/usr/include/cegui-0/CEGUI/CEGUI.h```.

**Comment by alkavan** [comment]
Here's an update. I tried the solution @hhyyrylainen offered.
It seems to work the best without the need adding ```/usr/local/include/cegui-0``` to global include path, the updated FindCEGUI.cmake from that repository did the job, in it seems it found couple of more stuff I did not see before during configuration.

**Comment by jjonj** [comment]
That find script won't work with the way we set things up on windows without some heavy modifications. I should be able to get the current one working!

**Comment by hhyyrylainen** [comment]
I need to potentially modify the find CEGUI script to add support for finding different versions.
As Ogre 2.0 requires CEGUI default branch (version number 9999). I already found a way to define the version with the official script but it should be relatively easy to modify the current one. But it would most likely conflict with any fixes to the include paths/I would have to later fix those, too.

**Comment by jjonj** [comment]
I already created a pull request for a fix, just needs to confirm that if it does indeed work on linux. #300

**Comment by jjonj** [comment]
Should have been fixed with #300
